### PR TITLE
PROD-678 View page of the Elementor saved template not working - Fixed

### LINKED ISSUE #3194: View page of the Elementor saved template not working
**Describe the bug**
View page of the Elementor saved template not working when news feed is set to as site home page. It is showing news feed page when clicking on the view link of the Elementor saved template. Please see this video: https://share.vidyard.com/watch/4fZwRF9yT6PTFPQpumtXsn?

**To Reproduce**
1. Make sure the News feed has been set as home page.
2. Go to Templates > Saved Templates
3. Click on the view for any saved templates

**Expected behavior**

**Environment**
Local, demo, client's site

**Support ticket links**
https://secure.helpscout.net/conversation/1703261813/174794/

**Jira issue** : [PROD-678]

[PROD-678]: https://buddyboss.atlassian.net/browse/PROD-678?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### PR DESCRIPTION
**Pull request by mahedihasannoman**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #3194  .

### How to test the changes in this Pull Request:

1. Make sure the News feed has been set as home page.
2. Go to Templates > Saved Templates
3. Click on the view for any saved templates
4. See it workds

### Proof Screenshots or Video

https://share.vidyard.com/watch/x7xJERGzLaU17oA9ZKEfRM?

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

File: src/bp-core/compatibility/bp-incompatible-plugins-helper.php [hunk 1]
```diff
@@ -790,6 +790,25 @@ function bb_get_elementor_maintenance_mode_template() {
...
+ */
+function bb_support_elementor_library_preview_permalink( $bp, $bp_uri ) {
+	global $bp_plugins;
+	if ( in_array( 'elementor/elementor.php', $bp_plugins ) && isset( $_GET['elementor_library'] ) ) {
+		$bp->current_component = '';
+		$bp->current_action    = '';
+	}
+}
+add_action( 'bp_core_set_uri_globals', 'bb_support_elementor_library_preview_permalink', 10, 2 );
+
```
Comment (chetansatasiya) [inline src/bp-core/compatibility/bp-incompatible-plugins-helper.php:811]: I would recommend creating a separate file for the elementor related compatibility issue same we did for other plugins.

**Review by chetansatasiya — CHANGES_REQUESTED** [review]
Need separate file for the elementor.

File: src/bp-core/compatibility/bp-incompatible-plugins-helper.php (hunk 1 above)
Comment (mahedihasannoman) [reply src/bp-core/compatibility/bp-incompatible-plugins-helper.php]: Hi @chetansatasiya , I have created a new file for elementor and moved all elementor compatibility related code there.

**Review by dipskakadiya — APPROVED** [review]

File: src/bp-core/compatibility/bp-incompatible-plugins-helper.php [hunk 2]
```diff
@@ -43,6 +43,15 @@ function bp_helper_plugins_loaded_callback() {
 		require buddypress()->compatibility_dir . '/bp-rankmath-plugin-helpers.php';
 	}
 
+	/**
+	 * Include plugin when plugin is activated
+	 *
+	 * Support Rank Math SEO
```
Comment (chetansatasiya) [inline src/bp-core/compatibility/bp-incompatible-plugins-helper.php:49]: @mahedihasannoman Is this related to Rank Math SEO?

**Review by chetansatasiya — CHANGES_REQUESTED** [review]
Need changes

File: src/bp-core/compatibility/bp-incompatible-plugins-helper.php (hunk 2 above)
Comment (mahedihasannoman) [reply src/bp-core/compatibility/bp-incompatible-plugins-helper.php]: Changed.

**Review by KartikSuthar — DISMISSED** [review]

**Pull request merged**